pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -351,14 +368,24 @@ func populateInitialImageStatus(ctx *domainContext, dirName string) {
 		size := int64(0)
 		info, err := os.Stat(filelocation)
 		if err != nil {
-			log.Error(err)
+			log.Error("Error in getting file information: ", err)
 		} else {
 			size = info.Size()
 		}
-		log.Debugf("populateInitialImageStatus: Processing %d Mbytes %s \n",
-			size/(1024*1024), filelocation)
+		_, sha256, appUuidStr := parseAppRwImageName(filelocation)
+		log.Debugf("populateInitialImageStatus: Processing AppUuid: %s, "+
+			"%d Mbytes, fileLocation:%s",
+			appUuidStr, size/(1024*1024), filelocation)
+
+		appUUID, err := uuid.FromString(appUuidStr)
+		if err != nil {
+			log.Errorf("populateInitialImageStatus: Invalid UUIDStr%s). err: %s,",
```

Comment:
What appUUID are you publishing in ImageStatus in this case? Either force it to a known value like the all-zeros UUID, or don't publish ImageStatus?

Reply:
The APP UUID is encoded in ihe image location - isn't it? I am extracting the App UUID from the image location.. A couple of lines above.. This is the Right App UUID - isn't it?

Reply:
My point is that you to a log.Errorf and then drive on as nothing happened. I don't know what thge return value is from uuid.FromString when it returns an error. Could be random garbage.

Reply:
Any update?

Reply:
Fixed error handling.. In case of such unexpected errors, delete the file and do not publish ImageStatus